lib/spack/spack/test/util/editor.py

```diff
@@ -132,3 +132,17 @@ def assert_exec(exe, args):
         assert args == [vim_exe, '/path/to/file']
 
     ed.editor('/path/to/file', _exec_func=assert_exec)
+
+
+def test_no_editor():
+    if 'VISUAL' in os.environ:
+        del os.environ['VISUAL']
+    if 'EDITOR' in os.environ:
+        del os.environ['EDITOR']
+    os.environ['PATH'] = ''
```

Comment:
It would be better to use `monkeypath.delenv` and `monkeypatch.setenv`, see [here](https://docs.pytest.org/en/7.1.x/how-to/monkeypatch.html)

Reply:
All the other tests in the suite all do it this way. The test infrastructure for Spack seems to save and restore the global environment.

Reply:
Fair. This is because of the `working_env` fixture set at the file level, which effectively duplicates part of what `monkeypatch` already does by default. I'm not convinced we should use it, but I'll probably create a tech0debt issue to remove that in another PR.